ASWF/CONTRIBUTING.md

```diff
@@ -0,0 +1,333 @@
+signatory power) who has the ability to use the online system to add or
+delete individual employees of the company who are authorized to submit pull
+requests, without needing to get an executive to amend and sign the
+agreement each time.
+
+Please note that these CLAs are based on the Apache 2.0 CLAs, and differ
+minimally, only as much as was required to correctly describe the EasyCLA
+process and our use of a CLA manager.
+
+**Contribution sign off**
+
+ASWF requires the use of the [Developer’s Certificate of Origin 1.1
+(DCO)](https://developercertificate.org/), which is the same mechanism that
+the [Linux®
+Kernel](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/submitting-patches.rst#n416)
+and many other communities use to manage code contributions. The DCO is
+considered one of the simplest tools for sign offs from contributors as the
+representations are meant to be easy to read and indicating signoff is done
+as a part of the commit message.
+
+Here is an example Signed-off-by line, which indicates that the submitter
+accepts the DCO:
+
+    Signed-off-by: John Doe <john.doe@example.com>
+
+You can include this automatically when you commit a change to your local
+git repository using `git commit -s`. You might also want to
+leverage this [command line tool](https://github.com/coderanger/dco) for
+automatically adding the signoff message on commits.
+
+
+Pull Requests and Code Review
+-----------------------------
+
+The best way to submit changes is via GitHub Pull Request. GitHub has a
+[Pull Request Howto](https://help.github.com/articles/using-pull-requests/).
+
+All code must be formally reviewed before being merged into the official
+repository. The protocol is like this:
+
+1. Get a GitHub account, fork OpenImageIO/oiio to create your own repository
+on GitHub, and then clone it to get a repository on your local machine.
+
+1. Edit, compile, and test your changes. Run clang-format (see the
+instructions on coding style below).
+
+1. Push your changes to your fork (each unrelated pull request to a separate
+"topic branch", please).
+
+1. Make a "pull request" on GitHub for your patch.
+
+2. If your patch will induce a major compatibility break, or has a design
+component that deserves extended discussion or debate among the wider OIIO
+community, then it may be prudent to email oiio-dev pointing everybody to
+the pull request URL and discussing any issues you think are important.
+
+1. All pull requests automatically launch CI jobs on GitHub Actions to
+ensure that the build completes and that the tests suite runs correctly, for
+a variety of platform, compiler, library, and flag combinations. The status
+of the CI tests for your PR will be displayed on the GitHub PR page. We will
+not accept PRs that don't build cleanly or pass the existing testsuite.
+
+1. The reviewer will look over the code and critique on the "comments" area.
```

Comment:
I think we should clarify if the submitter can be their own reviewer and if they can merge their own code into master. Are there some devs that should have this ability? Maybe the Chief Architect?

Reply:
I'm going to merge what we have when we get the rest of the votes in (since technically it already has the majority approval). Then I'll make a separate PR that takes a stab at what I think I can do to tighten it up, but we should discuss to get consensus (in the next meeting, if we can't do it easily on the PR). 

But I will add some of my initial thoughts here to get the ball rolling. In all cases, this is not set in stone, these are just my ideas based on other projects and what I think will make things efficient and careful.

The reviewer should definitely be somebody other than the patch author. I think once reviewed, either the reviewer or the patch author can be the one to push the "merge" button. Ideally it should be the patch author in cases where that person has commit privileges, but when that's not the case, any onlooker with commit privileges can do the actual merge if they are satisfied that the review is sufficient.

I would also like to be stricter about every patch being reviewed, versus my current begrudgingly accepted, but necessary practice of merging my own patches without review if they sit for a few days without any comments. I don't want to do that anymore, but what I need to make that work is for the TSC members and committers to make a serious effort to step in with the reviews and not allow any PRs to fester without review.

A few exceptions should be noted: 
* Some kinds of low-risk changes that can't break anything (like changes to documentation or administrative files) or high-priority fixes to critical issues that provably fix something very broken (such as a change to the CI scripts that change us from failing to passing CI) are ok for a senior developer to merge before getting a review if integration is time sensitive as long as a PR was submitted and CI fully passes.
* Some changes are so extensive or fundamental (especially backward-incompatible changes to APIs or behavior) that it is prudent to get reviews from multiple people, or representatives of multiple companies. There are no formal guidelines for this, we should just watch out for situations where multiple people or more time to review would be helpful for big changes.
* Backports of a reviewed/approved/CI-passing patch from master to a release branch can be done by the chief architect or release manager without a separate PR/review cycle just for the backport.